src/main/java/at/hannibal2/skyhanni/utils/Matrix.kt

```diff
@@ -0,0 +1,208 @@
+        )
+    }
+
+    operator fun set(index: Int, value: DoubleArray) {
+        this.data[index] = value
+    }
+
+    fun copy(): Matrix {
+        val newDoubleArray: Array<DoubleArray> = Array(height) {
+            DoubleArray(width)
+        }
+        for (rowIndex in data.indices) {
+            val row = data[rowIndex]
+            for (colIndex in row.indices) {
+                newDoubleArray[rowIndex][colIndex] = row[colIndex]
+            }
+        }
+        return Matrix(
+            newDoubleArray,
+        )
+    }
+
+    operator fun times(other: Double): Matrix {
+        val newDoubleArray: Array<DoubleArray> = Array(height) {
+            DoubleArray(width)
+        }
+        for (rowIndex in data.indices) {
+            val row = data[rowIndex]
+            for (colIndex in row.indices) {
+                val cell = row[colIndex]
+                newDoubleArray[rowIndex][colIndex] = cell * other
+            }
+        }
+        return Matrix(
+            newDoubleArray,
+        )
+    }
+
+    operator fun plus(other: Matrix): Matrix {
+        if (other.width != this.width || other.height != this.height) throw Error("Invalid Dimensions")
+        val newDoubleArray: Array<DoubleArray> = Array(height) {
+            DoubleArray(width)
+        }
+        for (rowIndex in data.indices) {
+            val row = data[rowIndex]
+            for (colIndex in row.indices) {
+                val cell = row[colIndex]
+                newDoubleArray[rowIndex][colIndex] = cell + other[rowIndex][colIndex]
+            }
+        }
+        return Matrix(
+            newDoubleArray,
+        )
+    }
+
+    operator fun minus(other: Matrix): Matrix {
+        if (other.width != this.width || other.height != this.height) throw Error("Invalid Dimensions")
+        val newDoubleArray: Array<DoubleArray> = Array(height) {
+            DoubleArray(width)
+        }
+        for (rowIndex in data.indices) {
+            val row = data[rowIndex]
+            for (colIndex in row.indices) {
+                val cell = row[colIndex]
+                newDoubleArray[rowIndex][colIndex] = cell - other[rowIndex][colIndex]
+            }
+        }
+        return Matrix(
+            newDoubleArray,
+        )
+    }
+
+    override fun toString(): String {
+        var output = ""
+        for (rowIndex in data.indices) {
+            val row = data[rowIndex]
+            for (colIndex in row.indices) {
+                val col = row[colIndex]
+                output += " $col"
+            }
+            output += "\n"
+        }
+        return output
+    }
+
+    companion object {
+        fun identity(rows: Int): Matrix {
+            val newDoubleArray: Array<DoubleArray> = Array(rows) {
+                DoubleArray(rows)
+            }
+            for (i in 0 until rows) {
+                newDoubleArray[i][i] = 1.0
+            }
+            return Matrix(
+                newDoubleArray,
+            )
```

Comment:
```suggestion
            return Matrix(newDoubleArray)
```
same thing with other places in the class when this is done